src/zcash/Note.cpp

```diff
@@ -137,3 +137,115 @@ ZCNoteEncryption::Ciphertext SproutNotePlaintext::encrypt(ZCNoteEncryption& encr
 
     return encryptor.encrypt(pk_enc, pt);
 }
+
+
+
+// Construct and populate SaplingNotePlaintext for a given note and memo.
+SaplingNotePlaintext::SaplingNotePlaintext(
+    const SaplingNote& note,
+    std::array<unsigned char, ZC_MEMO_SIZE> memo) : BaseNotePlaintext(note, memo)
+{
+    d = note.d;
+    rcm = note.r;
+}
+
+
+boost::optional<SaplingNote> SaplingNotePlaintext::note(const SaplingIncomingViewingKey& ivk) const
+{
+    auto addr = ivk.address( d );
```

Comment:
are the spaces in the parentheses intentional?

Reply:
No, will fix.